### PR TITLE
Add msg parameter to Equals function in testutil

### LINKED ISSUE #390: Add a "msg"  parameter to testutil.Equals.
In most cases we end up doing something like:

```
// Blocks should have been created .....
testutil.Equals(t, 0, len(db.blocks))
```
Instead it would be better if we have something like:

```
testutil.Equals(t, 0, len(db.blocks), "Blocks didn't get created")
```
This would help when looking at the output of the  failing tests.

### PR DESCRIPTION
**Pull request by janickic**
Fixes #390 
I can go through and attempt to add more meaningful messages as opposed to an empty string.

Signed-off-by: Camille Janicki <camille.janicki@gmail.com>

### CODE REVIEW

**Comment by krasi-georgiev** [comment]
yeah I would say lets add some messages so I can test it locally and see how it would look like with passing and failing tests.
Thanks!

File: wal_test.go [hunk 1]
```diff
@@ -58,16 +58,16 @@ func TestSegmentWAL_cut(t *testing.T) {
...
 
 		// We cannot actually check for correct pre-allocation as it is
 		// optional per filesystem and handled transparently.
 		et, flag, b, err := newWALReader(nil, nil).entry(f)
 		testutil.Ok(t, err)
-		testutil.Equals(t, WALEntrySeries, et)
-		testutil.Equals(t, byte(walSeriesSimple), flag)
-		testutil.Equals(t, []byte("Hello World!!"), b)
+		testutil.Equals(t, WALEntrySeries, et, "")
+		testutil.Equals(t, flag, byte(walSeriesSimple), "")
```
Comment (codesome) [inline wal_test.go]: here

**Comment by csmarchbanks** [comment]
I took a slightly different approach in a prometheus/prometheus PR (https://github.com/prometheus/prometheus/pull/4639). It might be worthwhile to choose one or the other for people working in both repos.

**Comment by krasi-georgiev** [comment]
Had a quick look and looks good. I would say let's take your approach

File: db_test.go [hunk 2]
```diff
@@ -110,7 +110,7 @@ func TestDataAvailableOnlyAfterCommit(t *testing.T) {
 	testutil.Ok(t, err)
 	seriesSet := query(t, querier, labels.NewEqualMatcher("foo", "bar"))
 
-	testutil.Equals(t, map[string][]sample{}, seriesSet)
+	testutil.Equals(t, seriesSet, map[string][]sample{}, "")
```
[the remark this answers is not in the document]
Comment (janickic) [reply db_test.go]: Will fix these. It looks like my code editor swapped the values

**Comment by janickic** [comment]
@csmarchbanks Added you as a co-author because I used your code.

**Comment by krasi-georgiev** [comment]
ping me when it is ready for another review.

**Comment by janickic** [comment]
@krasi-georgiev It looks like the checks are passing now. Thanks @krasi-georgiev and @codesome for suggesting to rebase.

**Comment by krasi-georgiev** [comment]
@codesome can you have another look and will merge.

File: testutil/testutil.go [hunk 3]
```diff
@@ -58,10 +58,20 @@ func NotOk(tb testing.TB, err error) {
 }
 
 // Equals fails the test if exp is not equal to act.
-func Equals(tb testing.TB, exp, act interface{}) {
+func Equals(tb testing.TB, exp, act interface{}, msgAndArgs ...interface{}) {
 	if !reflect.DeepEqual(exp, act) {
 		_, file, line, _ := runtime.Caller(1)
-		fmt.Printf("\033[31m%s:%d:\n\n\texp: %#v\n\n\tgot: %#v\033[39m\n\n", filepath.Base(file), line, exp, act)
+		fmt.Printf("\033[31m%s:%d:\n\n\texp: %#v\n\n\tgot: %#v%s\033[39m\n\n", filepath.Base(file), line, exp, act, formatMessage(msgAndArgs))
```
[the remark this answers is not in the document]
Comment (krasi-georgiev) [reply testutil/testutil.go]: yes tested it locally and you are right.

```
fmt.Printf("\033[31m%s:%d:%s\n\n\texp: %#v\n\n\tgot: %#v\033[39m\n\n", filepath.Base(file), line, formatMessage(msgAndArgs), exp, act)
```

File: testutil/testutil.go [hunk 4]
```diff
@@ -58,10 +58,20 @@ func NotOk(tb testing.TB, err error) {
...
 }
+
+func formatMessage(msgAndArgs []interface{}) string {
+	if len(msgAndArgs) == 0 {
+		return ""
+	}
+	if _, ok := msgAndArgs[0].(string); !ok {
+		return ""
+	}
+	return fmt.Sprintf(fmt.Sprintf("\n\nmsg: %s", msgAndArgs[0]), msgAndArgs[1:]...)
```
[the remark this answers is not in the document]
Comment (krasi-georgiev) [reply testutil/testutil.go]: +1

**Comment by janickic** [comment]
Thanks @codesome for the review!

**Comment by krasi-georgiev** [comment]
Thanks! appreciated.

**Comment by janickic** [comment]
I'll look into it! Thanks! :)

**Pull request merged**